test/e2e/aks.go

```diff
@@ -208,3 +209,74 @@ func WaitForAKSSystemNodePoolMachinesToExist(ctx context.Context, input WaitForC
 		return false
 	}, intervals...).Should(Equal(true), "System machine pools not detected")
 }
+
+func getAzureCluster(ctx context.Context, managementClusterClient client.Client, namespace, name string) (*infrav1.AzureCluster, error) {
+	key := client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	azCluster := &infrav1.AzureCluster{}
+	err := managementClusterClient.Get(ctx, key, azCluster)
+	return azCluster, err
+}
+
+func getAzureManagedControlPlane(ctx context.Context, managementClusterClient client.Client, namespace, name string) (*infrav1exp.AzureManagedControlPlane, error) {
```

Comment:
Indeed, I was just trying to not add a weird looking func (there are already a bunch w/ a well-established return pattern).

How about this, I'll remove the retry stuff from this PR, and submit a follow-up PR that turns all of these into retryable (and asserts on a terminal error, rather than returns an err).